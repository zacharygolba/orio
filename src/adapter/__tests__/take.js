// @flow

import Take from '../take'
import * as iter from '../../'

let subj

beforeEach(() => {
  const { producer } = iter.repeat('test')
  subj = new Take(producer, 3)
})

test('#@@iterator()', () => {
  for (const item of subj) {
    expect(item).toMatchSnapshot()
  }
})

test('#next()', () => {
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
})
