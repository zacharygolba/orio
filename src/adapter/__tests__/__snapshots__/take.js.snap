// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#@@iterator() 1`] = `"test"`;

exports[`#@@iterator() 2`] = `"test"`;

exports[`#@@iterator() 3`] = `"test"`;

exports[`#next() 1`] = `
Object {
  "done": false,
  "value": "test",
}
`;

exports[`#next() 2`] = `
Object {
  "done": false,
  "value": "test",
}
`;

exports[`#next() 3`] = `
Object {
  "done": false,
  "value": "test",
}
`;

exports[`#next() 4`] = `
Object {
  "done": true,
  "value": undefined,
}
`;
